pallets/compliance-manager/src/lib.rs

```diff
@@ -267,12 +273,14 @@ decl_module! {
                 rule_id: Self::get_latest_rule_id(ticker) + 1u32
             };
 
-            <AssetRulesMap>::mutate(ticker, |old_asset_rules| {
-                if !old_asset_rules.rules.iter().position(|rule| rule.sender_rules == new_rule.sender_rules && rule.receiver_rules == new_rule.receiver_rules).is_some() {
-                    old_asset_rules.rules.push(new_rule.clone());
-                    Self::deposit_event(Event::NewAssetRuleCreated(did, ticker, new_rule));
-                }
-            });
+            let mut asset_rules = <AssetRulesMap>::get(ticker);
+
+            if !asset_rules.rules.iter().any(|rule| rule.sender_rules == new_rule.sender_rules && rule.receiver_rules == new_rule.receiver_rules) {
+                asset_rules.rules.push(new_rule.clone());
+                Self::verify_rules_complexity(&asset_rules.rules, <TrustedClaimIssuer>::get(ticker).len())?;
```

Comment:
@maxsam4 Some time ago I saw that [DecodeLength](https://substrate.dev/rustdocs/v2.0.0-rc4/parity_scale_codec/trait.DecodeLength.html). If we used it for `<TrustedClaimIssuer>::get(ticker)`, we would improve the performance because we won't need to fetch the whole list.
However I am not sure if we could use it or if that is easy to use at this level.

Reply:
Nice find Miguel! It's pretty straightforward to use, pushing an update now.